Backpack-SwiftUI/SponsoredBanner/Classes/BPKSponsoredBanner.swift

```diff
@@ -0,0 +1,201 @@
+    let customAccessibilityLabel: String?
+    
+    @State var showTerms: Bool
+    
+    public init(
+        logo: Image? = nil,
+        title: String? = nil,
+        subheadline: String? = nil,
+        callToAction: CallToAction? = nil,
+        bodyText: String? = nil,
+        variant: Variant,
+        backgroundColor: Color,
+        showTerms: Bool = false,
+        customAccessibilityLabel: String? = nil
+    ) {
+        self.logo = logo
+        self.title = title
+        self.subheadline = subheadline
+        self.callToAction = callToAction
+        self.bodyText = bodyText
+        self.variant = variant
+        self.backgroundColor = backgroundColor
+        self.showTerms = showTerms
+        self.customAccessibilityLabel = customAccessibilityLabel
+    }
+    
+    public var body: some View {
+        VStack(spacing: 0) {
+            topView
+                .background(backgroundColor)
+                .zIndex(1)
+            bodyTextView
+            
+        }
+        .background(
+            RoundedRectangle(cornerRadius: .sm, style: .continuous)
+                .foregroundColor(.sponsoredBannerBackgroundColor)
+        )
+        .clipShape(
+            RoundedRectangle(cornerRadius: .sm, style: .continuous)
+        )
+        .if(customAccessibilityLabel != nil) { view in
+            view
+                .accessibilityElement(children: .ignore)
+                .accessibilityLabel(customAccessibilityLabel ?? "")
+        }
+    }
+    
+    private var bodyTextView: some View {
+        Group {
+            if hasBody && showTerms == true {
+                BPKText(bodyText ?? "", style: .caption)
+                    .lineLimit(4)
+                    .frame(maxWidth: .infinity)
+                    .padding(.base)
+                    .background(Color.clear)
+                    .zIndex(0)
+            }
+        }
+    }
+    
+    private var hasBody: Bool {
+        return bodyText != nil
+    }
+    
+    private var topView: some View {
+        Button(action: toggleBodyText) {
+            buttonContent
+        }
+        .accessibilityRemoveTraits(hasBody ? [] : .isButton)
+        .accessibilityElement(children: .combine)
+        .buttonStyle(SponsoredBannerButtonStyle(
+            foregroundColor: (variant == .onDark) ? Color(BPKColor.textOnDarkColor) : Color(BPKColor.textOnLightColor),
+            backgroundColor: backgroundColor
+        ))
+    }
+    
+    private var buttonContent: some View {
+        HStack(spacing: .md) {
+            if let logo = logo {
+                logo
+                    .resizable()
+                    .scaledToFit()
+                    .frame(maxHeight: 22)
+            }
+            titlesView
+            Spacer()
+            if let callToAction = callToAction {
+                BPKText(callToAction.text, style: .caption)
+                    .foregroundColor(variant.color)
+            }
+            if callToAction?.showIcon == true {
```

Comment:
should this line be inside the if let thats right above?

Reply:
Good catch!